src/Orleans/Runtime/AsynchQueueAgent.cs

```diff
@@ -111,6 +112,23 @@ public override void Stop()
             base.Stop();
         }
 
+        protected void DrainQueue(Action<T> action)
+        {
+            while (true)
+            {
+                T request;
+                try
+                {
+                    request = requestQueue.Take();
+                }
+                catch (InvalidOperationException)
+                {
+                    break; // Queue is empty
```

Comment:
Yes I guess it should be better to just use TryTake. But in this case, should I check that `Stop()` was called before?